entity-service-impl/src/main/java/org/hypertrace/entity/type/service/v2/model/EntityTypeDocument.java

```diff
@@ -0,0 +1,117 @@
+package org.hypertrace.entity.type.service.v2.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Objects;
+import org.hypertrace.core.documentstore.Document;
+import org.hypertrace.entity.service.constants.EntityServiceConstants;
+import org.hypertrace.entity.type.service.v2.EntityType;
+
+public class EntityTypeDocument implements Document {
+  // Since there could be additional metadata fields written by the doc store,
+  // ignore them while reading the object from JSON.
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().configure(
+      DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);;
+
+  @JsonProperty(value = EntityServiceConstants.TENANT_ID)
+  private String tenantId;
+
+  @JsonProperty(value = EntityServiceConstants.NAME)
+  private String name;
+
+  @JsonProperty
+  private String attributeScope;
+
+  @JsonProperty
+  private String idAttributeKey;
+
+  public EntityTypeDocument() {}
+
+  public EntityTypeDocument(String name, String attributeScope, String idAttributeKey) {
+    this(null, name, attributeScope, idAttributeKey);
```

Comment:
null tenantId? that seems bad - should we require the tenantId in the fromProto call (or a request context or something?)

Reply:
Exactly thought the same. Fixed.